### DISCUSSION TITLE
Preprocessing Data - PostgresML

### DISCUSSION THREAD

**Discussion by giscus** [postgresml/postgresml#547] [Announcements]
# Preprocessing Data - PostgresML

Train and deploy models to make online predictions using only SQL, with an open source Postgres extension.

https://postgresml.org/user_guides/training/preprocessing/

**Comment by dusanmarjanovic** [comment]
From what I tried this should be adjusted so that scale is "scale" in every lane and to separate each argument for preprocess with ,

SELECT pgml.train(
    project_name => 'preprocessed_model', 
    task => 'classification', 
    relation_name => 'weather_data',
    target => 'rain', 
    preprocess => '{
        "month":    {"encode": {"ordinal": ["jan", "feb", "mar", "apr", "may", "jun", "jul", "aug", "sep", "oct", "nov", "dec"]}}
        "clouds":   {"encode": "target", scale: "standard"}
        "humidity": {"impute": "mean", scale: "standard"}
        "temp":     {"scale": "standard"}
    }'
);
